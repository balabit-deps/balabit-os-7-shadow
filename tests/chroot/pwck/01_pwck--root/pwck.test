#!/bin/sh

set -e

cd $(dirname $0)

. ../../../common/config.sh
. ../../../common/log.sh

log_start "$0" "pwck can change a chroot"

save_config

# restore the files on exit
trap 'log_status "$0" "FAILURE"; clean_chroot; restore_config' 0

change_config

prepare_chroot

echo -n "pwck in a chroot (pwck --read-only --root $PWD/tmp/root)..."
pwck --read-only --root $PWD/tmp/root >tmp/pwck.out && exit 1 || {
        status=$?
}
echo "OK"

echo -n "Check returned status ($status)..."
test "$status" = "2"
echo "OK"

echo "pwck reported:"
echo "======================================================================="
cat tmp/pwck.out
echo "======================================================================="
echo -n "Check that there were a failure message..."
diff -au data/pwck.out tmp/pwck.out
echo "error message OK."
rm -f tmp/pwck.out

echo -n "Check the passwd file..."
../../../common/compare_file.pl config/etc/passwd /etc/passwd
../../../common/compare_file.pl config_chroot/etc/passwd tmp/root/etc/passwd
echo "OK"
echo -n "Check the group file..."
../../../common/compare_file.pl config/etc/group /etc/group
../../../common/compare_file.pl config_chroot/etc/group tmp/root/etc/group
echo "OK"
echo -n "Check the shadow file..."
../../../common/compare_file.pl config/etc/shadow /etc/shadow
../../../common/compare_file.pl config_chroot/etc/shadow tmp/root/etc/shadow
echo "OK"
echo -n "Check the gshadow file..."
../../../common/compare_file.pl config/etc/gshadow /etc/gshadow
../../../common/compare_file.pl config_chroot/etc/gshadow tmp/root/etc/gshadow
echo "OK"

rm -f tmp/root/etc/.pwd.lock
rm -f tmp/root/etc/passwd-
rm -f tmp/root/etc/group-
rm -f tmp/root/etc/shadow-
rm -f tmp/root/etc/gshadow-
clean_chroot

log_status "$0" "SUCCESS"
restore_config
trap '' 0

