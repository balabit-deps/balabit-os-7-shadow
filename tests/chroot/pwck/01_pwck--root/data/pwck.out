user 'root': program '/bin/bash' does not exist
user 'daemon': directory '/usr/sbin' does not exist
user 'daemon': program '/bin/sh' does not exist
user 'bin': directory '/bin' does not exist
user 'bin': program '/bin/sh' does not exist
user 'sys': directory '/dev' does not exist
user 'sys': program '/bin/sh' does not exist
user 'sync': directory '/bin' does not exist
user 'sync': program '/bin/sync' does not exist
user 'games': directory '/usr/games' does not exist
user 'games': program '/bin/sh' does not exist
user 'man': directory '/var/cache/man' does not exist
user 'man': program '/bin/sh' does not exist
user 'lp': directory '/var/spool/lpd' does not exist
user 'lp': program '/bin/sh' does not exist
user 'mail': directory '/var/mail' does not exist
user 'mail': program '/bin/sh' does not exist
user 'news': directory '/var/spool/news' does not exist
user 'news': program '/bin/sh' does not exist
user 'uucp': directory '/var/spool/uucp' does not exist
user 'uucp': program '/bin/sh' does not exist
user 'proxy': directory '/bin' does not exist
user 'proxy': program '/bin/sh' does not exist
user 'www-data': directory '/var/www' does not exist
user 'www-data': program '/bin/sh' does not exist
user 'backup': directory '/var/backups' does not exist
user 'backup': program '/bin/sh' does not exist
user 'list': directory '/var/list' does not exist
user 'list': program '/bin/sh' does not exist
user 'irc': directory '/var/run/ircd' does not exist
user 'irc': program '/bin/sh' does not exist
user 'gnats': directory '/var/lib/gnats' does not exist
user 'gnats': program '/bin/sh' does not exist
user 'nobody': directory '/nonexistent' does not exist
user 'nobody': program '/bin/sh' does not exist
user 'Debian-exim': directory '/var/spool/exim4' does not exist
user 'Debian-exim': program '/bin/false' does not exist
user 'myuser': directory '/home/' does not exist
user 'myuser': program '/bin/sh' does not exist
duplicate password entry
delete line 'testsuite::424243:424243::/home:/bin/bash'? No
user 'testsuite': no group 424243
user 'testsuite': directory '/home' does not exist
user 'testsuite': program '/bin/bash' does not exist
no matching password file entry in /etc/shadow
add user 'testsuite' in /etc/shadow? No
duplicate password entry
delete line 'testsuite::424244:424244::/home:/bin/bash'? No
user 'testsuite': no group 424244
user 'testsuite': directory '/home' does not exist
user 'testsuite': program '/bin/bash' does not exist
no matching password file entry in /etc/shadow
add user 'testsuite' in /etc/shadow? No
user 'testsuite1': no group 424243
user 'testsuite1': directory '/home' does not exist
user 'testsuite1': program '/bin/bash' does not exist
no matching password file entry in /etc/shadow
add user 'testsuite1' in /etc/shadow? No
pwck: no changes
